### PR TITLE
Fix CUDA tests on master

### PR DESCRIPTION
**Pull request by neerajprad**

### CODE REVIEW

File: pyro/ops/contract.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 def _finfo(tensor):
     # This can be replaced with torch.finfo once it is available
     # https://github.com/pytorch/pytorch/issues/10742
-    return np.finfo(torch.empty(torch.Size(), dtype=tensor.dtype).numpy().dtype)
+    return np.finfo(torch.empty(torch.Size(), dtype=tensor.dtype).cpu().numpy().dtype)
```
Comment (fritzo) [inline pyro/ops/contract.py]: let's avoid GPU here:
```py
np.finfo(torch.empty(torch.Size(), dtype=tensor.dtype, device='cpu').numpy().dtype)
```

File: tests/infer/test_valid_models.py [hunk 2]
```diff
@@ -1300,6 +1301,7 @@ def guide():
 
 
 @pytest.mark.parametrize('markov', [False, True])
+@pytest.mark.skipif("CUDA_TEST" in os.environ, reason="https://github.com/uber/pyro/issues/1380")
```
Comment (fritzo) [inline tests/infer/test_valid_models.py]: This issue is already resolved. If this is still failing, try changing `range(20)` to `range(15)`.

Comment (neerajprad) [reply tests/infer/test_valid_models.py]: I had to use`range(11)` - anything above that fails.

Comment (fritzo) [reply tests/infer/test_valid_models.py]: Oh I see, that makes sense since each time step allocates a new dim, so `range(11)` should allocate 22 dims.

**Comment by neerajprad** [comment]
This should be good to merge.

**Pull request merged**